lib/data/file_manager/file_manager.dart

```diff
@@ -390,18 +396,36 @@ class FileManager {
         .map((FileSystemEntity entity) {
           String filePath = entity.path.substring(documentsDirectory.length);
 
-          // remove extension
-          if (filePath.endsWith(Editor.extension)) {
-            filePath = filePath.substring(
-                0, filePath.length - Editor.extension.length);
-          } else if (filePath.endsWith(Editor.extensionOldJson)) {
-            filePath = filePath.substring(
-                0, filePath.length - Editor.extensionOldJson.length);
-          }
-
+          //log.info('Listing file $filePath');
           if (Editor.isReservedPath(filePath))
-            return null; // filter out reserved files
-
+            return null; // filter out reserved files - always
+
+          if (removeExtension){
+            // remove extension - return only names of notes (not assets)
+            if (filePath.endsWith(Editor.extension)) {
+              filePath = filePath.substring(
+                  0, filePath.length - Editor.extension.length);
+            } else if (filePath.endsWith(Editor.extensionOldJson)) {
+              filePath = filePath.substring(
+                  0, filePath.length - Editor.extensionOldJson.length);
+            }
+            else {
+              return null; // filePath is name of some asset
+            }
+          }
+          else {
+            if (includeAssetFiles){
```

Comment:
nit: I think you can remove a layer of if nesting.

Disclaimer: I'm not involved in this project but happened along the PR when trying to debug the same issue. So thanks for making a fix! Noticed a couple nits but again I'm just a random SWE so feel free to ignore.

I could be off here but IIUC this block is primary meant to return early to avoid including asset files when the argument is false If so it may be simpler to pull this if check into the inner most else down below and remove a layer of nesting.

We may be also be able to flip the if check to return the path if valid vs. return early if not. Given the arguments here it may simplify things.

I.e. something like:

```
if (includeAssetFiles || (is old note) || (is new note)){ return ... add file }
else { return null }
```

The main advantages being a little easier to read conditional (although that is a bit of an opinion on my part) and reduced nesting.

Reply:
If removeExtension is false, then there are two possibilities: 
1. includeAssetFiles is true, then every file name should be returned. 
2. if includeAssetFiles is false, then only file names sbn and sbn2 should be returned.

In fact are now used only two combinations:

removeExtension=true and includeAssetFiles=false (original code which lists only names of notes)
removeExtension=false and includeAssetFiles=true (code used in case of resync all which returns every file name)

So the code for removeExtension=true and includeAssetFiles=true  which are you discussing will never be interpreted and it will not be slow.

I think that it is OK and I would leave the decision on @adil192